arch/arm/include/armv7-m/irq.h

```diff
@@ -573,7 +573,7 @@ int up_cpu_index(void) noinstrument_function;
 
 static inline_function uint32_t up_getsp(void)
 {
-  register uint32_t sp;
```

Comment:
add register macro to compiler.h or add empty define in GHS case

Reply:
regiser keyword is deprecated by standard, we should drop all reference from code base.

Reply:
If other open source codes also use the `register` keyword, will you also go to their community to tell them that your writing is wrong?

Reply:
but the problem source file come from nuttx, we need fix it anyway.
whether add register macro in compiler.h is an another topic.

Reply:
so I think this is better to implement this topic instead of changing the kernel code. The `register` keyword will remind readers of the special meaning of this variable.

Reply:
@extinguish let's map register to empty in compiler.h for Greenhill toolchain.